docs/admin/architectures/index.md

```diff
@@ -0,0 +1,343 @@
+  - Evaluate the workspace utilization pattern. For instance, web application
+    development does not require high CPU capacity at all times, but will spike
+    during builds or testing.
+  - Evaluate minimal limits for single workspace. Include in the calculation
+    requirements for Coder agent running in an idle workspace - 0.1 vCPU and 256
+    MB. For instance, developers can choose between 0.5-8 vCPUs, and 1-16 GB
+    memory.
+
+#### Scaling formula
+
+When determining scaling requirements, consider the following factors:
+
+- `1 vCPU x 2 GB memory x 1 workspace`: A formula to determine resource
+  allocation based on the minimal requirements for an idle workspace with a
+  running Coder agent and occasional CPU and memory bursts for building
+  projects.
+
+**Node Autoscaling**
+
+Workspace nodes can be set to operate in autoscaling mode to mitigate the risk
+of prolonged high resource utilization.
+
+One approach is to scale up workspace nodes when total CPU usage or memory
+consumption reaches 80%. Another option is to scale based on metrics such as the
+number of workspaces or active users. It's important to note that as new users
+onboard, the autoscaling configuration should account for ongoing workspaces.
+
+Scaling down workspace nodes to zero is not recommended, as it will result in
+longer wait times for workspace provisioning by users. However, this may be
+necessary for workspaces with special resource requirements (e.g. GPUs) that
+incur significant cost overheads.
+
+### Data plane: External database
+
+While running in production, Coder requires a access to an external PostgreSQL
+database. Depending on the scale of the user-base, workspace activity, and High
+Availability requirements, the amount of CPU and memory resources required by
+Coder's database may differ.
+
+#### Scaling formula
+
+When determining scaling requirements, take into account the following
+considerations:
+
+- `2 vCPU x 8 GB RAM x 512 GB storage`: A baseline for database requirements for
+  Coder deployment with less than 1000 users, and low activity level (30% active
+  users). This capacity should be sufficient to support 100 external
+  provisioners.
+- Storage size depends on user activity, workspace builds, log verbosity,
+  overhead on database encryption, etc.
+- Allocate an additional CPU core to the database instance for every 1000 active
+  users.
+- Enable _High Availability_ mode for database engine for large scale
+  deployments.
+
+If you enable [database encryption](../encryption.md) in Coder, consider
+allocating an additional CPU core to every `coderd` replica.
+
+#### Performance optimization guidelines
+
+We provide the following general recommendations for PostgreSQL settings:
+
+- Increase number of vCPU if CPU utilization or database latency is high.
+- Allocate extra memory if database performance is poor and CPU utilization is
+  low. For maximum performance, the entire database should be able to fit in
+  RAM.
```

Comment:
I think we may want to drop the "fit in to RAM" statement, as it isn't necessarily true, fast storage can be just as good (if not better).

Perhaps better would be to simply state that "if memory utilization is high, and cpu low, then more RAM".

Reply:
Yep, good point.